### PR TITLE
Adds a stamina meter

### PR DESCRIPTION
**Pull request by DGamerL**
## What Does This PR Do
Adds a stamina meter to the right of the screen
Ports the icons from https://github.com/tgstation/tgstation/pull/64011 , as they were much better than my codersprite

**The stamina meter is invisible at 0 staminaloss**

## Why It's Good For The Game
Gives a bit of extra visual clarity if you are being stunned, and to separate staminacrit from actual crit in the health HUD

TODO:
- [x] Remove stamina damage from the normal health HUD (oops)
- [x] Account for painkillers

Now, one concern I do have, and others might aswell, is that this would make it a bit clearer if you are being hit by tirizine, or just a poison. I'll gladly accept design/balance team feedback on this

## Images of changes
ALL IMAGES BELOW EXCEPT THE LAST HAVE THE WRONG POSITIONING
0 Staminaloss
image

30 Staminaloss
image

60 Staminaloss
image

90 Staminaloss
image

Stamcrit
image

Now disconnected from the main damage HUD
image

New position - ALL THE ABOVE HAVE THE WRONG POSITION
image

## Changelog
:cl: DGamerL, That REALLY Good Soda Flavor
add: Added a stamina bar to the right of the screen
/:cl:

### CODE REVIEW

**Review by Contrabang — COMMENTED** [review]
IMO this should go below the `healths` hud to prevent more cramming of the status effect column

**Comment by pismorizzo** [comment]
Could it possibly start off green as well so it fits nicely with the body and overall health huds

**Comment by Daylight2** [comment]
I really like the current system of not being able to tell exactly how injured you are when hit by a combo of normal attacks and stuns. Seems realistic. But if we must have this, could it perhaps be invisible whilst at 100%, similar to the O2 warning? Even for super combat oriented roles like nukeops, they'll spend less than 1% of their time affected by stamina damage.

File: code/modules/mob/living/living_life.dm [hunk 1]
```diff
@@ -180,3 +180,23 @@
 			throw_alert("succumb", /atom/movable/screen/alert/succumb)
 		else
 			clear_alert("succumb")
+
+/mob/living/update_stamina_hud()
+	if(!client)
+		return
+	if(staminas)
```
Comment (lewcc) [inline code/modules/mob/living/living_life.dm:187]: early return here

File: code/modules/mob/living/carbon/carbon_life.dm [hunk 2]
```diff
@@ -285,6 +286,27 @@
 		else
 			healths.icon_state = "health7"
 
+/mob/living/carbon/update_stamina_hud()
+	if(!client)
+		return
+	if(staminas)
+		var/perceived_stamina = staminaloss - shock_reduction()
```
Comment (lewcc) [inline code/modules/mob/living/carbon/carbon_life.dm:293]: It feels to me like you could probably add a proc to determine the perceived stamina on the `mob/living` level (and override it here on the carbon level), and then just remove this override in favor of a single implementation

Comment (DGamerL) [reply code/modules/mob/living/carbon/carbon_life.dm]: Apologies fort he long wait, done!

File: code/modules/mob/living/living_life.dm (hunk 1 above)
Comment (DGamerL) [reply code/modules/mob/living/living_life.dm]: Done!

**Comment by DGamerL** [comment]
> I really like the current system of not being able to tell exactly how injured you are when hit by a combo of normal attacks and stuns. Seems realistic. But if we must have this, could it perhaps be invisible whilst at 100%, similar to the O2 warning? Even for super combat oriented roles like nukeops, they'll spend less than 1% of their time affected by stamina damage.

Did this!

**Review by Burzah — APPROVED** [review]

**Review by dearmochi — APPROVED** [review]

**Pull request merged**